### PR TITLE
#343 Global provider variable.

### PR DESCRIPTION
**Pull request by criske**
PR for #343

### CODE REVIEW

**Comment by charlesmike** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by criske** [comment]
@amihaiemil no bounty

File: src/main/resources/templates/head.html [hunk 1]
```diff
@@ -26,14 +26,23 @@
...
     <script type="text/javascript">
-        /*
-           @todo #335:60min. We should have a global js variable `provider` that will be initialized when
-            `/api/users/self` call in `head.html` is successful (`provider = data.provider`).
-            This variable should replace thymeleaf directives that uses `${provider}` passed as model attribute.
-            Since this operation is async, we should somehow observe changes applied to this variable.
-            Basically when `provider` variable is changed, use jQuery to update dom elements where thymeleaf ${provider} is used.
-            A good starting point to achieve this, would be using proxies: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy#manipulating_dom_nodes
-        */
+        window.globalProvider = (function () {
```
Comment (criske) [inline src/main/resources/templates/head.html:29]: @amihaiemil added to the window scope, to ensure that this is a global variable.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2021-01-24T13:46:13.734719``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

**Comment by criske** [comment]
@amihaiemil  I've noticed that we call  `user/self` again in `getProject.js` just to get `user.login`
https://github.com/self-xdsd/self-web/blob/45a6b726662a33edf164825f458364c32de53e86/src/main/resources/public/js/getProject.js#L10-L14

I think we could use the same aproach for a `globalProfile` (this will include the provider too)
```javascript
window.globalProfile.onChange(function(user){
   $.get("/api/projects/"+owner+"/"+name, function(project) {
      displayProject(user.login, project);
   }
})
```

**Comment by charlesmike** [comment]
@amihaiemil Don't forget to close this ticket before the deadline (2021-01-24T13:46:13). You are past the first half of the allowed period.

**Comment by amihaiemil** [comment]
@criske looks good. Can you open a ticket for the user profile page? :D

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**